src/platform_impl/android/mod.rs

```diff
@@ -103,7 +103,13 @@ pub struct EventLoop {
     window_target: ActiveEventLoop,
     redraw_flag: SharedFlag,
     loop_running: bool, // Dispatched `NewEvents<Init>`
-    running: bool,
+    // /// `onStart()` - `onStop()` lifecycle, denoting when the app is visible
+    // TODO: Might use activity_visible for allowing proxy wakeup events?
+    // activity_visible: bool,
+    // /// `onResume()` - `onPause()` lifecycle, denoting when the app is actively interacted with
+    // activity_active: bool,
+    // TODO: This should be per-Activity (likely analogous to per-Window) state.
+    window_has_surface: bool,
```

Comment:
I started with these states just to see what would happen, but it's quite common to have:

- `onStart()`
- `onResume()`
- `NativeWindowCreated`
- `NativeWindowResized`
- `NativeWindowRedrawNeeded`
- `WindowFocusChanged` to `1`
- ...
- `WindowFocusChanged` to `0`
- `onPause()`
- `NativeWindowDestroyed`
- `onStop()`
- (`SaveInstanceState`...)